app/ui-react/packages/api/src/useIntegrationHelpers.tsx

```diff
@@ -343,12 +346,16 @@ export const useIntegrationHelpers = () => {
           action,
           configuredProperties,
           connection,
-          id: key(),
-          metadata: { configured: true } as any,
+          id: originalStep ? originalStep.id : key(),
+          metadata: {
+            ...(originalStep ? originalStep.metadata : {}),
+            ...{ configured: true },
+          } as any,
           stepKind: 'endpoint',
         },
         actionDescriptor!
       );
+      console.log('new', step);
```

Comment:
Some left over debug logging?

Reply:
🤦‍♂️ thanks, I had that changed but forgot to save.